### PR TITLE
mkosi: Build a sysext if SYSEXT=1 is specified

### PR DESCRIPTION
**Pull request by daandemeyer**

### CODE REVIEW

File: mkosi.functions [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+Type=root
+Format=erofs
+CopyFiles=/usr/
+Minimize=best
+EOF
+
+    mkdir -p "$1"/usr/lib/extension-release.d
+    cp /usr/lib/os-release "$1"/usr/lib/extension-release.d/extension-release.systemd
+    cat >>"$1"/usr/lib/extension-release.d/extension-release.systemd <<EOF
+SYSEXT_ID=systemd
```
Comment (bluca) [inline mkosi.functions:22]: this also needs ID= matching the image it is being built for

Comment (daandemeyer) [reply mkosi.functions:22]: @bluca See the line just above this

Comment (bluca) [reply mkosi.functions:22]: ah it's copied verbatim - mmh we usually don't want to copy the full thing, maybe grep ID and (if present) VERSION_ID or SYSEXT_LEVEL from it instead? That's how we usually build these in the CI

**Review by bluca — APPROVED** [review]

File: mkosi.functions [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    if [[ -n "$SYSEXT_LEVEL" ]]; then
+        cat >>"$1"/usr/lib/extension-release.d/extension-release.systemd <<EOF
+SYSEXT_LEVEL=$SYSEXT_LEVEL
+EOF
+    fi
+
+    rm -f "$BUILDDIR"/systemd.raw
+
+    env --unset=SYSTEMD_REPART_OVERRIDE_FSTYPE_ROOT \
+    "$BUILDDIR"/systemd-repart \
```
Comment (bluca) [inline mkosi.functions:42]: maybe sign it too, if mkosi.key is there?

Comment (daandemeyer) [reply mkosi.functions:42]: Let's do that later, we should do it properly and parse the key and crt location from the mkosi config but I can't be bothered to implement that atm

**Pull request merged**